extensions/amp-story/1.0/amp-story.js

```diff
@@ -1148,6 +1158,21 @@ export class AmpStory extends AMP.BaseElement {
           this.getPageIndex(oldPage) < pageIndex ?
             setAttributeInMutate(oldPage, Attributes.VISITED) :
             removeAttributeInMutate(oldPage, Attributes.VISITED);
+
+          if (oldPage.isAd()) {
+            this.storeService_.dispatch(
+                Action.SET_ADVANCEMENT_MODE,
+                AdvancementMode.ADVANCE_TO_ADS);
+          }
+
+          // TODO(alanorozco): check if autoplay
+          this.navigationState_.updateActivePage(
+              pageIndex,
+              this.getPageCount(),
+              targetPage.element.id,
+              oldPage.element.id,
+              targetPage.getNextPageId() === null /* isFinalPage */
+          );
```

Comment:
Moving this piece of code here changes the behavior of the analytics: now it won't get triggered for the first page, when a user loads the story.
I think we can keep the code where it was before, and just make sure we don't cause any error when oldPage is not defined.

Reply:
Got it & done.